### PR TITLE
feat: rewrite e2e test(server_info_test) with ginkgo

### PR DESCRIPTION
**Pull request by Jaycean**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues
#1500

___
### New feature or improvement
-  Rewrite e2e test(server_info) with ginkgo

### CODE REVIEW

**Comment by liuxiran** [comment]
a strange ci failed, just re-run to see the results

**Review by starsz — APPROVED** [review]

**Comment by nic-chen** [comment]
> a strange ci failed, just re-run to see the results

hi, @liuxiran  which ci failed ? I think we should fix it

**Comment by Jaycean** [comment]
> > a strange ci failed, just re-run to see the results
> 
> hi, @liuxiran which ci failed ? I think we should fix it

this ci failed, Test building web in multiple node version / build (14.x) (pull_request)

**Comment by nic-chen** [comment]
> this ci failed, Test building web in multiple node version / build (14.x) (pull_request)

ok, thanks.

File: api/test/e2enew/server_info/server_info_test.go [hunk 1]
```diff
@@ -14,127 +14,113 @@
...
 			Method:       http.MethodGet,
-			Headers:      map[string]string{"Authorization": token},
+			Headers:      map[string]string{"Authorization": base.GetToken()},
 			ExpectStatus: http.StatusOK,
 			ExpectBody:   "\"hostname\":\"apisix_server1\"",
-		},
-		{
-			Desc:         "get server info",
-			Object:       ManagerApiExpect(t),
+			Sleep:        base.SleepTime * 15,
```
Comment (nic-chen) [inline api/test/e2enew/server_info/server_info_test.go:40]: why sleep so long ..

**Comment by Jaycean** [comment]
> > this ci failed, Test building web in multiple node version / build (14.x) (pull_request)
> 
> ok, thanks.

I saw in the morning that CI failed without any error message

File: api/test/e2enew/server_info/server_info_test.go (hunk 1 above)
Comment (Jaycean) [reply api/test/e2enew/server_info/server_info_test.go:40]: Old code wait for APISIX report, sleep  2 seconds... So I sleep a little longer
```
	// wait for apisix report
	time.Sleep(2 * time.Second)
```

Comment (nic-chen) [reply api/test/e2enew/server_info/server_info_test.go:40]: @Jaycean 
It is recommended to directly use the original time - 2 seconds.

Some places may rely on services that are delayed. For example, here is the reporting interval of the plugin `server-info`.

Comment (Jaycean) [reply api/test/e2enew/server_info/server_info_test.go]: done.

**Review by nic-chen — APPROVED** [review]

**Comment by nic-chen** [comment]
merged, thanks a lot! @Jaycean

**Pull request merged**